src/FactSystem/FactMetaData.cc

```diff
@@ -1012,66 +1012,32 @@ const FactMetaData::AppSettingsTranslation_s* FactMetaData::_findAppSettingsHori
             continue;
         }
 
-        uint settingsUnits = qgcApp()->toolbox()->settingsManager()->unitsSettings()->horizontalDistanceUnits()->rawValue().toUInt();
-
-        if (pAppSettingsTranslation->unitType == UnitHorizontalDistance
-                && pAppSettingsTranslation->unitOption == settingsUnits) {
-            return pAppSettingsTranslation;
-        }
-    }
-    return nullptr;
-}
-
-const FactMetaData::AppSettingsTranslation_s* FactMetaData::_findAppSettingsVerticalDistanceUnitsTranslation(const QString& rawUnits)
-{
-    for (size_t i=0; i<sizeof(_rgAppSettingsTranslations)/sizeof(_rgAppSettingsTranslations[0]); i++) {
-        const AppSettingsTranslation_s* pAppSettingsTranslation = &_rgAppSettingsTranslations[i];
-
-        if (rawUnits.toLower() != pAppSettingsTranslation->rawUnits.toLower()) {
-            continue;
+        uint settingsUnits = 0;
+        switch (type) {
+        case UnitHorizontalDistance:
+            settingsUnits = qgcApp()->toolbox()->settingsManager()->unitsSettings()->horizontalDistanceUnits()->rawValue().toUInt();
```

Comment:
Ok, just one last thing please. Can you move `qgcApp()->toolbox()->settingsManager()->unitsSettings()` into a variable as well to deal with the same repetition leads to bugs problem.

Reply:
Absolutely agree, looks much better: https://github.com/mavlink/qgroundcontrol/pull/8820/commits/78cdbbbd0e2d3ef718a23b1ca5d0ae61481eda8d